rust/parquet/src/arrow/arrow_writer.rs

```diff
@@ -144,12 +145,17 @@ fn write_leaves(
         | ArrowDataType::Time32(_)
         | ArrowDataType::Time64(_)
         | ArrowDataType::Duration(_)
-        | ArrowDataType::Interval(_) => {
+        | ArrowDataType::Interval(_)
+        | ArrowDataType::LargeBinary
+        | ArrowDataType::Binary
+        | ArrowDataType::Utf8
+        | ArrowDataType::LargeUtf8 => {
             let mut col_writer = get_col_writer(&mut row_group_writer)?;
             write_leaf(
                 &mut col_writer,
                 array,
                 levels.pop().expect("Levels exhausted"),
+                array.data_type(),
```

Comment:
you need not pass the data type as you can get it from the array

Reply:
:+1: